mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/NativeQueryParser.java

```diff
@@ -63,12 +65,14 @@ public Rule CriteriaOnlyFindQuery() {
 
 	@SuppressNode
 	public Rule Db() {
-		return Sequence( ZeroOrMore( WhiteSpace() ), "db " );
+		return Sequence( ZeroOrMore( WhiteSpace() ), "db ", Separator() );
 	}
 
 	@SuppressSubnodes
 	public Rule Collection() {
-		return Sequence( OneOrMore( TestNot( Separator() ), ANY ), builder.setCollection( match() ) );
+		return Sequence( OneOrMore( TestNot( Reserved() ), ANY ), builder.setCollection( match() ) );
+		//TODO it should not be just ANY matcher as they are some restrictions in the Collection naming in Mongo
```

Comment:
Could you open a JIRA issue so we remember to implement these checks, too? Thanks